### PR TITLE
Don't return 500 if mirror url contains special chars

### LINKED ISSUE #31640: HTTP 500 error when adding push mirror with invalid character in URL
### Description

When I copied and pasted a URL for a push mirror it included a control character (in this case a tab).
After supplying a username and password, the user interface failed with an "Internal Server Error" (i.e. HTTP 500)

I tracked it down in the logs to be caused by this error:
```
2024/07/16 09:50:17 ...o/setting/setting.go:412:SettingsPost() [E] SanitizeURL: parse "https://github.com/<user redacted>/<repository redacted>\t": net/url: invalid control character in URL
2024/07/16 09:50:17 .../context_response.go:70:HTML() [D] Template: status/500
2024/07/16 09:50:17 ...eb/routing/logger.go:102:func1() [I] router: completed POST /<username redacted>/<repository redacted>/settings for [<IPv6 address redacted>]:0, 500 Internal Server Error in 76.2ms @ setting/setting.go:100(setting.SettingsPost)
```

I am wondering if the SettingsPost() method (specifically the SanitizeURL part) could be modified to trim off control characters from the start or end of what has been supplied as a URL?

### Gitea Version

1.22.1 built with GNU Make 4.4.1, go1.22.5 : bindata, timetzdata, sqlite, sqlite_unlock_notify

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

2.45.2

### Operating System

Docker container

### How are you running Gitea?

Docker container
`image: gitea/gitea:latest`

### Database

MySQL/MariaDB

### PR DESCRIPTION
**Pull request by lunny**
Fix #31640

### CODE REVIEW

**Comment by yp05327** [comment]
Mirror also has the same problem in L242.

**Comment by lunny** [comment]
> Mirror also has the same problem in L242.

d88d02ee27fe156992a73ab423270bf0c063604b

File: routers/web/repo/setting/setting.go [hunk 1]
```diff
@@ -401,7 +402,8 @@ func SettingsPost(ctx *context.Context) {
 
 		remoteAddress, err := util.SanitizeURL(form.PushMirrorAddress)
 		if err != nil {
-			ctx.ServerError("SanitizeURL", err)
+			ctx.Data["Err_PushMirrorAddress"] = true
+			ctx.RenderWithErr(ctx.Tr("repo.mirror_address_url_invalid"), tplSettingsOptions, &form)
```
Comment (yp05327) [inline routers/web/repo/setting/setting.go]: using `handleSettingRemoteAddrError` too ?

**Review by yp05327 — APPROVED** [review]

File: routers/web/repo/setting/setting.go (hunk 1 above)
Comment (lunny) [reply routers/web/repo/setting/setting.go]: 20d3dae8e79fdb009b213ae0a05238573542171a

**Review by Zettat123 — APPROVED** [review]

**Pull request merged**